packages/app/src/systems/CRX/background/services/BackgroundService.ts

```diff
@@ -181,7 +181,13 @@ export class BackgroundService {
   }
 
   async isConnected(_: unknown, serverParams: EventOrigin) {
-    return !!serverParams.connection;
+    const currentAccount = await AccountService.getCurrentAccount();
+    if (!serverParams.connection) {
+      return false;
+    }
+    return serverParams.connection.accounts.includes(
+      currentAccount?.address || ''
+    );
```

Comment:
We should change this implementation to add an optional param to isConnected.

The idea of the isConnected method is not connected to the current account but to the state of the connection.

Another way to know if an account is connected or not is by checking on the accounts list.

Reply:
The current behavior is not a bug but an expected behavior. cc: @LuizAsFight

Reply:
what's the benefit of knowing that the account is connected but you can't do nothing as the current account is not connected? on this `isConnected` has no use as you can't trust it to show "connect" button for example.

Reply:
You can request a signature or send transactions to any connected account.

Reply:
@matt-user we can close this PR then

Reply:
@LuizAsFight ok I'll close this PR and refactor the code on the bridge